### PR TITLE
LPS-140227 Disable the delete option in Entries linked to another one into a relationship with deletion type Prevent

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-140227

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java [hunk 1]
```diff
@@ -583,6 +585,29 @@ private ServiceContext _createServiceContext(
...
+						ObjectRelationshipConstants.TYPE_MANY_TO_MANY) &&
+					!objectRelationship.isReverse()) {
+
+					return objectRelationship;
+				}
+
+				return null;
+			});
+	}
+
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:610]: @danvb112 @marcelabc I think this method should be inside ObjectRelationshioLocalService. That way, we avoid processing over a list.

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java [hunk 2]
```diff
@@ -591,6 +616,85 @@ private String _getObjectEntryPermissionName(long objectDefinitionId) {
 		return ObjectDefinition.class.getName() + "#" + objectDefinitionId;
 	}
 
+	private boolean _hasPreventDeletionType(long objectDefinitionId)
+		throws PortalException {
+
+		_objectRelationships = new ArrayList<>();
+
+		_objectRelationships.addAll(
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:624]: @danvb112 @marcelabc should we be checking all relationships at once?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java [hunk 3]
```diff
@@ -591,6 +616,85 @@ private String _getObjectEntryPermissionName(long objectDefinitionId) {
...
+		}
+
+		return false;
+	}
+
+	private boolean _hasRelatedObjectEntries(
+			com.liferay.object.model.ObjectEntry objectEntry)
+		throws PortalException {
+
+		for (ObjectRelationship objectRelationship : _objectRelationships) {
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:655]: @marcelabc can we try to follow the same pattern you did for a similar use case? That one that we've consulted the DB directly?

**Review by gabrielwas — CHANGES_REQUESTED** [review]

**Comment by gabrielwas** [comment]
@danvb112 @marcelabc do you have any updates from that one?

**Comment by marcelabc** [comment]
Resent: https://github.com/liferay/liferay-portal/pull/4930

**Pull request closed without merging**